src/llvm/function_pass_manager.cr

```diff
@@ -1,6 +1,5 @@
-{% unless LibLLVM::IS_LT_130 %}
-  @[Deprecated("The legacy pass manager was removed in LLVM 17. Use `LLVM::PassBuilderOptions` instead")]
-{% end %}
```

Comment:
**issue:** This class is part of the stdlib API. We should not simply remove it when building for newer LLVM versions.
I suppose w can increase the condition for the deprecation message to `IS_LT_170`.

Reply:
I did that at first, to avoid the deprecation warnings when compiling Crystal, but LLVM 17 removed the feature, so the linker will fail to link against LLVM 17+ anyway. There's no way around the breaking change :disappointed:

Reply:
Good point

Reply:
@straight-shoota fixed by keeping the Deprecated annotation but now only warning for LLVM 17+ rather than

I did remove the functions from LibLLVM though, as they don't exist anymore + they don't show up in documentation. The deprecation warnings shall trigger and compilation will fail as soon as the semantic phase, instead of compiling everything just to have the linker fail.